### PR TITLE
Rollback use of localstorage that do not work on safari :(

### PR DESCRIPTION
**Pull request by sbilodeau**

### CODE REVIEW

**Review by blaisefonseca — APPROVED** [review]

File: app/views/signin.js [hunk 1]
```diff
@@ -27,12 +26,13 @@ define(['app', 'authentication'], function() {
...
-            else                  localStorage.removeItem('cbd_authentication_email');
 
-            if(success.data.expiration) localStorage.setItem("cbd_authentication_expiration", success.data.expiration);
-            else      localStorage.removeItem("cbd_authentication_expiration");
+            var expires = new Date();
+
+            expires.setDate(expires.getDate()+365);
+            
+            if($scope.rememberMe) $cookies.put   ('email', $scope.email, { path:'/', expires: expires, samesite: 'None', secure: true });
+            else                  $cookies.remove('email', { path:'/' });
```
Comment (blaisefonseca) [inline app/views/signin.js:35]: can you all add the expiration date to the cookie?
```
if(success.data.expiration) 
   $cookies.put("expiration", success.data.expiration, { path:'/', expires: new Date(success.data.expiration), samesite: 'None', secure: true });
else      
  $cookies.remove('expiration', { path:'/' });
            ```

File: app/views/signin.js [hunk 2]
```diff
@@ -27,12 +26,13 @@ define(['app', 'authentication'], function() {
...
-            if($scope.rememberMe) localStorage.setItem('cbd_authentication_email', $scope.email);
-            else                  localStorage.removeItem('cbd_authentication_email');
 
-            if(success.data.expiration) localStorage.setItem("cbd_authentication_expiration", success.data.expiration);
-            else      localStorage.removeItem("cbd_authentication_expiration");
+            var expires = new Date();
+
+            expires.setDate(expires.getDate()+365);
+            
+            if($scope.rememberMe) $cookies.put   ('email', $scope.email, { path:'/', expires: expires, samesite: 'None', secure: true });
```
Comment (blaisefonseca) [inline app/views/signin.js:34]: Technically only when the user explicitly checks the checkbox 😄

File: app/authorize.html [hunk 3]
```diff
@@ -2,29 +2,43 @@
...
 }
-function removeFromStorage(key) {
-    return localStorage.removeItem(key);
+function setCookie(cname, cvalue, exdays)
+{
+    if(cvalue===null || cvalue === undefined)
+        exdays = -365;
+    var parts = [cname + "=" + (cvalue||"")];
+    if(exdays) {
+        var d = new Date();
```
Comment (blaisefonseca) [inline app/authorize.html:20]: should be outside the if block

File: app/authorize-test.js [hunk 4]
```diff
@@ -70,34 +59,36 @@ function receiveMessage(event) {
...
-            else                          removeFromStorage('expiration'); //Delete from storage
+
+            if(message.authenticationToken) setCookie('authenticationToken', message.authenticationToken, message.expiration);
+            else                            setCookie('authenticationToken', null); //Delete the cookie
+
+            if(message.authenticationEmail)        setCookie('email', message.authenticationEmail, 365);
+            if(message.authenticationEmail===null) setCookie('email', null); //Delete the cookie
+
+            if(message.expiration)        setCookie('expiration', message.expiration, 365);
+            else                          setCookie('expiration', null); //Delete the cookie
```
Comment (blaisefonseca) [inline app/authorize-test.js:91]: can you use `message.expiration` instead of 365

**Review by blaisefonseca — CHANGES_REQUESTED** [review]
few comments

File: app/views/signin.js (hunk 1 above)
Comment (sbilodeau) [reply app/views/signin.js:35]: Done

**Review by randyhoulahan — APPROVED** [review]

**Pull request merged**